### PR TITLE
fix: make version number a number

### PR DESCRIPTION
**Pull request by s-ashwinkumar**

### CODE REVIEW

File: lib/smartcar.rb [hunk 1]
```diff
@@ -51,4 +50,15 @@ class BadRequestError < ExternalServiceError; end
   STOP_CHARGE = "STOP".freeze
   # Constant for units
   UNITS = [IMPERIAL,METRIC]
+
+  # Smartcar API version variable - defaulted to 1.0
+  @api_version = "1.0"
+
+  def self.set_api_version(version)
```
Comment (s-ashwinkumar) [inline lib/smartcar.rb:57]: ruby doc for these methods

File: lib/smartcar/user.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@ class User < Base
     # @param token [String] Access token
     #
     # @return [String] User ID
-    def self.user_id(token:, version: DEFAULT_API_VERSION)
+    def self.user_id(token:, version: Smartcar.get_api_version)
```
Comment (s-ashwinkumar) [inline lib/smartcar/user.rb:16]: we dont need to document these `version` parameters in readme, but this gives us a little more control to operate with multiple version if needed (looking forwarded to future version migrations slowly like maybe comatibility v3 etc etc). If we dont want to give that control we can get rid of this argument everywhere....

File: spec/smartcar/e2e/vehicle_spec.rb [hunk 3]
```diff
@@ -7,7 +7,7 @@
   subject { Smartcar::Vehicle }
   before(:context) do
     client = Smartcar::Oauth.new(AuthHelper.auth_client_params)
-    url = client.authorization_url
+    url = client.authorization_url({ force_prompt: true })
```
Comment (s-ashwinkumar) [inline spec/smartcar/e2e/vehicle_spec.rb:10]: can reorganize this and move it into the helper method so as to not repeat it in all tests...something for later.

**Review by gurpreetatwal — APPROVED** [review]

File: lib/smartcar/user.rb (hunk 2 above)
Comment (gurpreetatwal) [reply lib/smartcar/user.rb:16]: makes sense, our plan is to do something similar on the other SDKs as well

File: spec/smartcar/e2e/vehicle_spec.rb (hunk 3 above)
Comment (gurpreetatwal) [reply spec/smartcar/e2e/vehicle_spec.rb:10]: makes sense

**Pull request merged**